src/core/crash-handler.c

```diff
@@ -113,7 +113,10 @@ _noreturn_ static void crash(int sig, siginfo_t *siginfo, void *context) {
                                         log_struct(LOG_EMERG,
                                                    LOG_MESSAGE("Caught <%s> from PID "PID_FMT".", signal_to_string(sig), siginfo->si_pid),
                                                    "MESSAGE_ID=" SD_MESSAGE_CRASH_PROCESS_SIGNAL_STR);
-                        }
+                        } else
+                                log_struct(LOG_EMERG,
+                                           LOG_MESSAGE("Caught <%s>, from signal code %d.", signal_to_string(sig), siginfo->si_code),
```

Comment:
any chance you can use a table for the si codes?

i.e. use DEFINE_STRING_TABLE_LOOKUP() macro.

i.e. add to signal-util.h this:

```c
const char* signal_code_to_string(in c) _const_;
```

and then to signal-util.c:

```c
static const char* const signal_code_table[] = {
        [SI_USER]  = "user",
        [SI_QUEUE] = "queue",
        …
};

DEFINE_STRING_TABLE_LOOKUP_TO_STRING(signal_code, int);
```

and then use that in the log msg, just like you already use signal_to_string(), but on si_code